akka-http/src/main/scala/akka/http/scaladsl/server/directives/RouteDirectives.scala

```diff
@@ -82,6 +83,47 @@ trait RouteDirectives {
   def handle(handler: HttpRequest => Future[HttpResponse]): StandardRoute =
     { ctx => handler(ctx.request).fast.map(RouteResult.Complete)(ctx.executionContext) }
 
+  /**
+   * Handle the request using a function.
+   *
+   * @group route
+   */
+  def handleSync(handler: HttpRequest => HttpResponse): StandardRoute =
+    { ctx => FastFuture.successful(RouteResult.Complete(handler(ctx.request))) }
+
+  /**
+   * Handle the request using an asynchronous partial function.
+   *
+   * This directive can be used to include external components request processing components defined as PartialFunction
+   * (like those provided by akka-grpc) into a routing tree defined as routes.
+   *
+   * @param rejections The list of rejections to reject with if the handler is not defined for a request. By default a request
+   *                   is rejected with an empty list of rejections in that case, which is interpreted as "Not Found".
+   *
+   * @group route
+   */
+  def handlePF(handler: PartialFunction[HttpRequest, Future[HttpResponse]], rejections: Seq[Rejection] = Nil): StandardRoute =
```

Comment:
:+1:, just `handle` would also be kind of nice but perhaps too ambiguous

Reply:
It seems that would work. I think the original reason I put something else in here was to make it easier to document but maybe it's not a real problem to show both signatures on one page.

Reply:
I made that change, but now it doesn't work any more because the wrong overload is chosen, I guess because of the default parameter in one case. I'll see if that can be fixed by even more overloads.

Reply:
Added more overloads, so the overload resolution now works correctly ;)